crates/node_binding/src/resolver.rs

```diff
@@ -44,15 +44,21 @@ impl JsResolver {
     path: String,
     request: String,
   ) -> napi::Result<Either<JsResourceData, bool>> {
-    block_on(async move {
-      match self.resolver.resolve(Path::new(&path), &request).await {
-        Ok(rspack_core::ResolveResult::Resource(resource)) => {
-          Ok(Either::A(ResourceData::from(resource).into()))
-        }
-        Ok(rspack_core::ResolveResult::Ignored) => Ok(Either::B(false)),
-        Err(err) => Err(napi::Error::from_reason(format!("{err:?}"))),
+    block_on(self._resolve(path, request))
+  }
+
+  pub async fn _resolve(
```

Comment:
shouldn't pub if it's not intended for public api

Reply:
I agree but it's not trivial to call _resolve in resolve since _resolve use &self which is not movable which may needs move the logic to some common util function, but it seems not neat too

Reply:
we can do some refactor in following PR